dvc/scm/git/__init__.py

```diff
@@ -133,8 +133,18 @@ def _get_gitignore(self, path):
 
         return entry, gitignore
 
-    @staticmethod
-    def _ignored(entry, gitignore_path):
+    def _ignored(self, entry, gitignore_path):
+
+        # We want to check first if `entry` is already being ignored by the .gitignore file in the root dir.
+        entry_path = os.path.join(os.path.dirname(gitignore_path), entry)
+        from git.exc import GitCommandError
+        try:
+            self.repo.git.check_ignore(entry_path)
+            return True
+        except GitCommandError:
+            # If none of the paths passed to `check_ignore` are ignored, GitPython annoyingly raises an Exception
+            pass
+
         if os.path.exists(gitignore_path):
```

Comment:
Btw, we no longer need to manually read gitignore as if the entry is present there, `check_ignore` will tell us that.

Reply:
ah interesting, will remove those then

Reply:
ah interesting. Will change that